src/System.Private.CoreLib/shared/System/IO/Path.cs

```diff
@@ -824,5 +824,35 @@ internal static StringComparison StringComparison
                     StringComparison.OrdinalIgnoreCase;
             }
         }
+
+        /// <summary>
+        /// Trims the ending directory separator if present.
+        /// </summary>
+        /// <param name="path"></param>
+        public static string TrimEndingDirectorySeparator(string path) =>
+            EndsInDirectorySeparator(path) && !PathInternal.IsRoot(path.AsSpan()) ?
```

Comment:
Should this trim all superfluous directory separators?

E.g. `c:\hello\\\\\\\\\\\` -> `c:\hello` ?

Reply:
If I'm not mistaken first idea was expose "as is" but... /cc @JeremyKuhne

Reply:
I don't think we should. I don't think that need would be common and you can loop if you really need to.

Reply:
This makes me again wonder what is this API for. The top post in https://github.com/dotnet/corefx/issues/31570 says that it is to make path comparisons simpler. If you are normalizing paths for comparison, you would want to trim all superfluous slashes, no?

Reply:
Or maybe the name should be `TrimLastEndingDirectorySeparator` the method could mislead dev used to using `TrimEnd` api

Reply:
@JeremyKuhne thoughts?

Reply:
> If you are normalizing paths for comparison, you would want to trim all superfluous slashes, no?

The key usage of this is comparing paths that have gone through normalization (i.e. `Path.GetFullPath()`) as it is silly to compare paths that haven't been normalized (`C:\foo\\bar` is the same as `C:\foo/bar` for example). We don't, however, want to introduce bugs with roots for those that haven't called it: `@"C:\"` does not equal `@"C:"` on Windows. Same goes for `"/"` and `""` on Unix. Those shouldn't typically show up. There is, however, also a rarer use case where you need a path for a further P/Invoke. Windows does not accept trailing slashes, outside of the root path. You'll find it fails, trim it yourself, and introduce a bug. If you're lucky things will just fail when you're working with a root, if you're not so lucky you'll be working from the current directory instead of the root.

Reply:
Ok, make sense. This should be all mentioned in the docs for this API.

Reply:
PR is ready to review, I'm attempting to have a gree CI but it's not so easy 😣